[RHDEVDOCS-3027] ODC Release Notes, Known Issues, and Bug Fixes for 4.8 (#34411)
openshift/openshift-docs: release_notes/ocp-4-8-release-notes.adoc

> * This update fixes the sort order and color of `ImageManifestVuln` objects for the Quay Security Operator. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1942716[*BZ#1942716*])

Author: Will remove this update.

> * Previously, the *Topology* view showed an error or a blank page while loading because the OpenShift Samples Operator was not installed. So there were no templates to the OpenShift namespace. The current release fixes this issue. The *Topology* view now loads as expected. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1949810[BZ#1949810])

Reviewer: ```suggestion
* Previously, the *Topology* view showed an error or a blank page while loading because the OpenShift Samples Operator was not installed and there were no templates to the OpenShift namespace. The current release fixes this issue. The *Topology* view now loads as expected. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1949810[BZ#1949810])
```

* Previously, the word "Operators" was translated for Chinese users, but the plural translation resulted in a bad user experience. Now the word "Operators" is not translated. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1945818[*BZ#1945818*])

* Previously, an incorrect code was causing the `User` and `Group` details to show unrelated subjects. Now, a code has been added to filter by `User` or `Group`, so the `User` and `Group` details show related subjects. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1951212[*BZ#1951212*])

> * Previously, when you imported a devfile, the OpenShift web console ignored the `build guidance` placeholder container which provided the configuration for environment variables, ports and limits. The new deployment had a second container which could not start because the placeholder image could not be fetched and it missed the required configurations. Now, the `build guidance` container is dropped from the new deployment, and the container adds the environment variables, ports and limit configurations. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1952214[BZ#1952214])

Reviewer 2: Add serial comma after 'ports' in first and the last sentence.

Author: done

* Previously, the colors for the *Low* and *Medium* severity issues of the Image Manifest Vulnerabilities (IMVs) did not match the color representation shown in the (link:https://quay.io/[Quay.io]) interface. As a result, when the user changed the severity order of vulnerabilities to *High*, the IMVs ordered the issues incorrectly. This created confusion when reviewing the IMVs. The current release fixes this issue. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1942716[BZ#1942716])

* Previously, if the API server encountered a conflict while updating a resource quota, it sometimes failed to create a resource and returned a 409 status code. In that case, the client had to retry the API request. With this update, when the API server encounters this type of conflict, it retries creating the resource up to three times. This is often sufficient to create the resource. Otherwise, the API server displays an error in the console. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1920699[BZ#1920699])

> * Previously, pipelines created in the Git import flow for private repositories failed to run. This happened because the pipeline `ServiceAccount` did not use secrets created by the Git import flow for private Git repositories. With this update, you can add a secret name to the annotations of the pipeline `ServiceAccount` object, and add pipeline-specific annotations to the provided secret. As a result, pipelines for private Git repositories run successfully. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1970470[BZ#1970470])

Reviewer 2: Please change the second sentence to - This happened because the pipeline `ServiceAccount` object did not use...

Reviewer 2: This applies to PR 34416 too: https://github.com/openshift/openshift-docs/pull/34416/files#diff-14b6355bc5db567351b9e43a469c73396794e644c2dd1fc8523417f2f7b72c16R1303

Author: done

* Previously, an incorrect code was causing `User` and `Group` details to show unrelated subjects. Now, a code has been added to filter by `User` or `Group`, so the `User` and `Group` details show related subjects. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1951212[*BZ#1951212*])

> * Previously, when you imported a devfile, the Openshift web console ignored the `build guidance` placeholder container because the new deployment had a second container. The web console could not start the container because it could not fetch the placeholder image. As a result, the user container missed the additional configurations. Now, in the web console, the `build guidance` container is dropped from the new deployment, and the container adds the environment variable, ports, and limit configurations. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1952214[BZ#1952214])

Reviewer 2: Currently the meaning seems to have changed. 
Suggest: 
Previously, when you imported a devfile, the Openshift web console ignored the `build guidance` placeholder container which provided the configuration for environment variables, ports and limits. The new deployment had a second container which could not start because the placeholder image could not be fetched and it missed the required configurations. Now, the `build guidance` container is dropped from the new deployment, and the container adds the environment variables, ports, and limit configurations.
Please get @sanketpathak to verify this.

Reviewer 3: +1 to what @Preeticp said, that's correct,  the build guidance container is dropped and the console should only keep the environment variables, ports, and limit configurations from this container,

Reviewer 4: +1

> * Previously, the *Topology* view showed an error or a blank page. This happened because the OpenShift Samples Operator loads page templates to the OpenShift namespace, and was not installed. The current release fixes this issue. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1949810[BZ#1949810])

Reviewer: ```suggestion
* Previously, the *Topology* view showed an error or a blank page. This happened because the OpenShift Samples Operator, which loads page templates to the OpenShift namespace, was not installed. The current release fixes this issue. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1949810[BZ#1949810])
```

> * Previously, routes tied to the *Developer* perspective were not rendered. This produced 404 error messages when you refresh the Developer perspective. The current release fixes this issue by loading all the inactive routes. Now, the routes switch to the correct perspective before loading. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1929769[BZ#1929769])

Reviewer: ```suggestion
* Previously, routes tied to the *Developer* perspective were not rendered. This produced 404 error messages when you refreshed the Developer perspective. The current release fixes this issue by loading all the inactive routes. Now, the routes switch to the correct perspective before loading. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1929769[BZ#1929769])
```

Reviewer 2: Developer needs to be in bold here -  when you refreshed the Developer perspective.

> * Previously, the colors for the Low and Medium severity issues of the Image Manifest Vulnerabilities (IMVs) did not match the color representation shown in the (link:https://quay.io/[Quay.io]) interface. As a result, when the user changed the severity order of vulnerabilities to High, the IMVs ordered the issues incorrectly. This created confusion when reviewing the IMVs. The current release fixes this issue. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1942716[BZ#1942716])

Reviewer 2: Are High and Low UI elements? If yes, please make them bold.

Author: Done

> * Previously, the colors for the Low and Medium severity issues of the Image Manifest Vulnerabilities (IMVs) did not match the color representation shown in the (link:https://quay.io/[Quay.io]) interface. As a result, when the user changed the severity order of vulnerabilities to High, the IMVs ordered the issues incorrectly. This created confusion when reviewing the IMVs. The current release fixes this issue. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1942716[BZ#1942716])

Author: Yeah, remove the above one.

> * Previously, if it encountered a conflict while updating a resource quota, the API server sometimes failed to create a resource and returned a 409 status code. In that case, the client had to retry the API request. With this update, when the API server encounters this type of conflict, it retries creating the resource up to three times. This is often sufficient to create the resource. Otherwise, the API server displays an error in the console. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1920699[BZ#1920699])

Reviewer 2: Previously, if the API server encountered a conflict while updating a resource quota, it  sometimes failed to...

Author: Done

* Previously, *Pipelines* you created in the git import flow for private repos failed to run.

> This happened because *Pipeline* ServiceAccount did not use secrets created by git import flow for private git repositories. With this update, you can add a secret name to the pipeline ServiceAccount's annotations, and add pipeline-specific annotations to the provided secret. As a result, pipelines for private git repositories run successfully. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1970470[BZ#1970470])

Reviewer: ```suggestion
This happened because the *Pipeline* `ServiceAccount` object did not use secrets created by git import flow for private git repositories. With this update, you can add a secret name to the pipeline `ServiceAccount` object's annotations, and add pipeline-specific annotations to the provided secret. As a result, pipelines for private git repositories run successfully. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1970470[BZ#1970470])
```

Reviewer: Are Pipeline and Pipelines bold because they are UI elements?

Reviewer 2: repos/repositories
*Pipelines*/pipelines
git/Git in all the instances, please
add a secret name to the pipeline `ServiceAccount` object's annotations/add a secret name to the annotations of the pipeline `ServiceAccount` object

Reviewer 5: - did not use secrets created by git import flow ---> did not use secrets created by the git import flow
- pipeline `ServiceAccount` object's annotations ---> annotations of the Pipeline `serviceAccount` object (Or) annotations of the Pipeline ServiceAccount object

Please check and use the correct way on how to represent this ^.

Author: Done

* Previously, when users inserted a formatted YAML snippet in the YAML editor, the new selection did not match the new content in the snippet: the indentation was removed, and some random letters remained behind the selection. The current release fixes this issue. Now, the cursor remains in the position where it started and adds the missing indentation for the cursor end position. After inserting the YAML snippet, the new selection matches the new content. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1952545[BZ#1952545])

> * Previously, annotations were passed to the specification of knative services along with metadata. As a result, decorators were shown for associated revisions of knative services in the topology. The current release fixes this issue by passing annotations only to knative service metadata. Now, decorators are shown only for knative services in the topology and not for associated revisions.

Reviewer 2: Knative in all instance - K capital
only to knative service metadata./only to the knative service metadata.

Author: Done

> * Previously, annotations were passed to the specification of knative services along with metadata. As a result, decorators were shown for associated revisions of knative services in the topology. The current release fixes this issue by passing annotations only to knative service metadata. Now, decorators are shown only for knative services in the topology and not for associated revisions.

Reviewer 5: knative ---> Knative

Please update this change for all such instances.

Author: Done

(link:https://bugzilla.redhat.com/show_bug.cgi?id=1954959[BZ#1954959])

> * Previously, when selecting the *YAML* tab, the `metadata.managedFields` section was not collapsing immediately. This was due to an issue with the Form or YAML switcher for pages such as *Pipeline Builder* and *Edit HorizontalPodAutoscaler* (HPA). As a result, the part of the document where the user tried to type collapsed. The `metadata.managedFields` section remained as is, and the cursor was reset to the starting position to the top left of the YAML editor. The current release fixes this issue. Now, upon loading the YAML, the `metadata.managedFields` section is immediately collapsed.

Reviewer 2: was not collapsing/did not collapse
Form and YAML should be in bold
section collapses immediately.

Author: Done

+
For more information, see link:https://bugzilla.redhat.com/show_bug.cgi?id=1812212[*BZ#1812212*].

> * In {product-title} 4.8, if the user has the console open in multiple tabs, some sidebar links in the *Developer perspective* do not directly link to the project, and there is an unexpected change in the selected project. This will be resolved in a future release. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1839101[*BZ#1839101*])

Reviewer 2: *Developer perspective*/*Developer* perspective
This will be resolved in a future release. - seems redundant imho, take it or leave it :)

Author: Done
